blockchain/src/blockchain/verify.rs

```diff
@@ -197,6 +199,48 @@ impl Blockchain {
         Ok(())
     }
 
+    /// Verify that all the given equivocation proofs of a block are actually valid offenses.
+    pub fn verify_equivocation_proofs(
+        &self,
+        block: &Block,
+        txn: &DBTransaction,
+    ) -> Result<(), PushError> {
+        // We don't need to perform any checks if the given block is not a micro block.
+        let micro_block = match block {
+            Block::Macro(_) => return Ok(()),
+            Block::Micro(micro_block) => micro_block,
+        };
+
+        let body = micro_block
+            .body
+            .as_ref()
+            .expect("Block body must be present");
+
+        for equivocation_proof in &body.equivocation_proofs {
```

Comment:
What about `valid_at`? I probably missed it, but i do not remember seeing it being checked. To my understanding the validity of a block depends also on that.

Reply:
It's already being called today, no changes necessary:
https://github.com/nimiq/core-rs-albatross/blob/f220d3ef2318cfdd25ea406132f1a81016c008c8/blockchain/src/blockchain/push.rs#L90-L91
https://github.com/nimiq/core-rs-albatross/blob/f220d3ef2318cfdd25ea406132f1a81016c008c8/blockchain/src/blockchain/verify.rs#L23-L24
https://github.com/nimiq/core-rs-albatross/blob/f220d3ef2318cfdd25ea406132f1a81016c008c8/primitives/block/src/block.rs#L396-L398
https://github.com/nimiq/core-rs-albatross/blob/f220d3ef2318cfdd25ea406132f1a81016c008c8/primitives/block/src/micro_block.rs#L248-L254